js/output/pjs/pjs-ast-transforms.js

```diff
@@ -150,15 +152,50 @@ ASTTransforms.rewriteContextVariables = function(envName, context) {
                     // methods.  In that case, always rewrite it.
                     if (scopes.length === 1 || drawLoopMethods.includes(decl.id.name)) {
                         if (["Program", "BlockStatement", "SwitchCase"].includes(parent.type)) {
-                            return b.ExpressionStatement(
-                                b.AssignmentExpression(
-                                    b.MemberExpression(
-                                        b.Identifier(envName),
-                                        b.Identifier(decl.id.name)),
-                                    "=",
-                                    decl.init
-                                )
-                            );
+                            // Transform all NewExpressions to CallExpressions
+                            // Example:
+                            // var myObj = new Obj(1);
+                            //  ...will be transformed to...
+                            // __env__.myObj = __env__.PJSOutput.applyInstance(__env__.Obj, 'Obj')(1);
+                            if (["NewExpression"].includes(decl.init.type)) {
```

Comment:
Could you put this into its own separate AST transform? The reason being is that I'm working on some code to export pjs programs so that they can be run outside of live-editor and I want to be able to toggle this transform off when doing the export. You'll need to modify exec in pjs-code-injector.js to run the new transform before rewriteContextVariables.